common/models/facility.js

```diff
@@ -35,6 +36,9 @@ module.exports = Facility => {
   // Facility.validatesUniquenessOf('facility_code_dhis2');
   // Facility.validatesUniquenessOf('facility_code_openlmis');
 
+  // after save hook is getting called too many times
```

Comment:
I noticed that the handler was being called a lot of times. There were no obvious loops or causes at the time for I was only updating a single field on the model during the trial. Just assumed the framework calls the hook a number of times. First thing that came to mind was to add the <i>once</i> HoF. Not sure, perhaps I missed something. Open to suggestions.

Reply:
I get you. At the moment I have no clue as to why that is the case, perhaps since other models are attached to facility one, they could also be initiating the hook. We will just need to make efforts when testing the UI to make sure that all functionality is intact after using the _once_ HoF

Reply:
@caspater do you mind checking into this branch and making sure that the UI works okay. Ideally, we should re-introduce our development - staging and master branch setup.

Reply:
Do you think it makes sense to add the following to the mailer job:
```javascript
/**
 * Checks to see that the job has been fed all the right info or throws
 * @return {undefined}
 */
function runPreconditions() {
  Preconditions.checkArg(ENV.get('MHFR_EMAIL_WORKER_CRON_EXPR'), 'Please set MHFR_EMAIL_WORKER_CRON_EXPR env var.');
  Preconditions.checkArg(ENV.get('MHFR_EMAIL_USER'), 'Please set MHFR_EMAIL_USER env var.');
  Preconditions.checkArg(ENV.get('MHFR_EMAIL_PASSWORD')), 'Please set MHFR_EMAIL_PASSWORD env var.';
  Preconditions.checkArg(ENV.get('MHFR_MEDICAL_COUNCIL_EMAIL'), 'Please set MHFR_MEDICAL_COUNCIL_EMAIL env var.');
  Preconditions.checkArg(ENV.get('MHFR_FACILITIES_PATH'), 'Please set MHFR_FACILITIES_PATH env var.');
}
```
It checks that the required env vars have been set for the job or throws. Now since the job runs under the app then when it throws it'll prevent the app from starting until the env vars have been set. Do you think it neccessary?

Reply:
> I get you. At the moment I have no clue as to why that is the case, perhaps since other models are attached to facility one, they could also be initiating the hook. We will just need to make efforts when testing the UI to make sure that all functionality is intact after using the _once_ HoF

Think it should still work since it is essentially a <i>closure</i>. Still, definitely no problem in being safe.

Reply:
> Do you think it makes sense to add the following to the mailer job:
> 
> ```js
> /**
>  * Checks to see that the job has been fed all the right info or throws
>  * @return {undefined}
>  */
> function runPreconditions() {
>   Preconditions.checkArg(ENV.get('MHFR_EMAIL_WORKER_CRON_EXPR'), 'Please set MHFR_EMAIL_WORKER_CRON_EXPR env var.');
>   Preconditions.checkArg(ENV.get('MHFR_EMAIL_USER'), 'Please set MHFR_EMAIL_USER env var.');
>   Preconditions.checkArg(ENV.get('MHFR_EMAIL_PASSWORD')), 'Please set MHFR_EMAIL_PASSWORD env var.';
>   Preconditions.checkArg(ENV.get('MHFR_MEDICAL_COUNCIL_EMAIL'), 'Please set MHFR_MEDICAL_COUNCIL_EMAIL env var.');
>   Preconditions.checkArg(ENV.get('MHFR_FACILITIES_PATH'), 'Please set MHFR_FACILITIES_PATH env var.');
> }
> ```
> 
> It checks that the required env vars have been set for the job or throws. Now since the job runs under the app then when it throws it'll prevent the app from starting until the env vars have been set. Do you think it necessary?

I'd say yes. Actually for all the vars if you will i.e not specifically for the emailing setup.

Reply:
@danielmwakanema please not that those env vars are missing from the example.env file

Reply:
On master they are present.

Reply:
Still waiting for @danielmwakanema . Ideally code pushed into master should not be breaking.

Reply:
I have added the missing env var in the .env.example file. Breaking? What do you mean?

Reply:
Ooh, I meant it would be breaking without that env variable being taking into consideration. All good now.